src/lib-ssl-iostream/iostream-openssl-context.c

```diff
@@ -226,7 +226,9 @@ static int ssl_ctx_use_certificate_chain(SSL_CTX *ctx, const char *cert)
 		ret = 0;
 
 	if (ret != 0) {
+#ifdef HAVE_SSL_CTX_set_current_cert
 		SSL_CTX_select_current_cert(ctx, x);
+#endif
```

Comment:
I wonder if something breaks now because these aren't called, @cmouse ?

Reply:
I'm not sure, would have to test this. @catap would be nice if you could test out that:

local_name cert {
}

blocks work still correctly with SNI

and that e.g. drwetter's SSL checker gives clean bill of health .